### ISSUE TITLE
Iris/CentraLite contact sensor 3320-L1

### ISSUE THREAD

**Issue by apmon** [dresden-elektronik/deconz-rest-plugin#667]
I have a Iris Contact Sensor that shows up in deConz as a CentraLite 3320-L. However, it doesn't seem to get a name assigned or show up in the web api. Presumably the product code just needs to be added to the list?

<img width="479" alt="screen shot 2018-05-20 at 7 51 47 pm" src="https://user-images.githubusercontent.com/251447/42110941-1329dc54-7ba0-11e8-9466-ddce84175aa6.png">
<img width="479" alt="screen shot 2018-05-20 at 7 51 08 pm" src="https://user-images.githubusercontent.com/251447/42110942-134c316e-7ba0-11e8-9770-bb86e027ea04.png">
<img width="394" alt="screen shot 2018-05-20 at 7 49 44 pm" src="https://user-images.githubusercontent.com/251447/42110943-13609faa-7ba0-11e8-84f9-b05af8450064.png">
<img width="396" alt="screen shot 2018-05-20 at 7 49 19 pm" src="https://user-images.githubusercontent.com/251447/42110944-137746e2-7ba0-11e8-8d11-4fb831473eff.png">
<img width="395" alt="screen shot 2018-05-20 at 7 48 57 pm" src="https://user-images.githubusercontent.com/251447/42110945-138b50a6-7ba0-11e8-9196-1bb9dde0b1b6.png">
<img width="328" alt="screen shot 2018-05-20 at 7 48 44 pm" src="https://user-images.githubusercontent.com/251447/42110946-139ea868-7ba0-11e8-9f1c-5ec63d7b5100.png">

**Comment by apmon** [comment]
I tried adding it to the supported devices array, but it looks like I did something wrong?
I do now see it as "3320-L" in the deCONZ gui, and it (both the ZHAOpenClose and ZHATemperature) do now seem to be rest api. But both have the 3320-L name instead of OpenClose25 and Temperature26. It also looks like it isn't reporting status changes.

Do you have some guides of where exactly a new device needs to be added to in the code?

**Comment by ebaauw** [comment]
That is kinda tricky, I always find there’s one more place that I overlooked...  I usually search the code for a similar device’s vendor, modelid and cluster, and keep my fingers crossed.

Obviously, in the `supportedDevices` array near the top of de_webplugin.cpp and possibly in both `addSensorNode()` functions in the same file.  The same loginc needs to be in database.cpp for when restoring the sensor.  Then there’s binding.cpp to setup attribute reporting - if the sensor resources don’t update, this is probably missing, although I don’t think it’s needed for the _IAS zone_ cluster.  From the screenshot above, the _IAS zone_ hasn’t been enrolled, though.  I think that should happen automatically upon discovery, but you might try that manually through the deCONZ GUI.  I had some issues that the IAS zone type would be read too late during discovery, so I whitelisted the models While processing the _IAS zone_ cluster, but if the ZHAOpenClose resource is created, that’s not an issue.  Also you would need to translate the battery voltage into a percentage, as is done for some other devices as well.
